synapse/handlers/devicemessage.py

```diff
@@ -50,7 +51,14 @@ def __init__(self, hs: "HomeServer"):
             "m.direct_to_device", self.on_direct_to_device_edu
         )
 
-        self._device_list_updater = hs.get_device_handler().device_list_updater
+        if hs.config.worker.worker_app is None:
```

Comment:
a comment here wouldn't go amiss: "If we're on a worker, send device resync requests over to the master" or something.